tests/e2e/isbservice.go

```diff
@@ -347,36 +348,39 @@ func updateISBServiceRollout(
 	originalPipelineName, err := getPipelineName(Namespace, pipelineRolloutName)
 	Expect(err).ShouldNot(HaveOccurred())
 
-	updateISBServiceRolloutInK8S(isbServiceRolloutName, func(rollout apiv1.ISBServiceRollout) (apiv1.ISBServiceRollout, error) {
+	UpdateISBServiceRolloutInK8S(isbServiceRolloutName, func(rollout apiv1.ISBServiceRollout) (apiv1.ISBServiceRollout, error) {
 		rollout.Spec.InterStepBufferService.Spec.Raw = rawSpec
 		return rollout, nil
 	})
 
 	// both "dataLoss" fields and "recreate" fields have risk of data loss
 	dataLossRisk := dataLossFieldChanged || recreateFieldChanged
-
-	if upgradeStrategy == config.PPNDStrategyID {
+	if UpgradeStrategy == config.PPNDStrategyID {
 		if dataLossRisk {
 
-			document("Verify that in-progress-strategy gets set to PPND")
+			Document("Verify that in-progress-strategy gets set to PPND")
 			verifyInProgressStrategyISBService(Namespace, isbServiceRolloutName, apiv1.UpgradeStrategyPPND)
-			verifyInProgressStrategy(pipelineRolloutName, apiv1.UpgradeStrategyPPND)
-			verifyPipelinePaused(Namespace, pipelineRolloutName)
+			// if we expect the pipeline to be failed?
+			// short term workaround
+			if !strings.Contains(pipelineRolloutName, "failed") {
+				VerifyInProgressStrategy(pipelineRolloutName, apiv1.UpgradeStrategyPPND)
+				VerifyPipelinePaused(Namespace, pipelineRolloutName)
+			}
 
-			document("Verify that the pipelines are unpaused by checking the PPND conditions on ISBService Rollout and PipelineRollout")
+			Document("Verify that the pipelines are unpaused by checking the PPND conditions on ISBService Rollout and PipelineRollout")
 			Eventually(func() bool {
 				isbRollout, _ := isbServiceRolloutClient.Get(ctx, isbServiceRolloutName, metav1.GetOptions{})
 				isbCondStatus := getRolloutConditionStatus(isbRollout.Status.Conditions, apiv1.ConditionPausingPipelines)
 				plRollout, _ := pipelineRolloutClient.Get(ctx, pipelineRolloutName, metav1.GetOptions{})
 				plCondStatus := getRolloutConditionStatus(plRollout.Status.Conditions, apiv1.ConditionPipelinePausingOrPaused)
-				if isbCondStatus != metav1.ConditionTrue || plCondStatus != metav1.ConditionTrue {
+				if isbCondStatus == metav1.ConditionTrue || plCondStatus == metav1.ConditionTrue {
```

Comment:
glad you caught that one. :) I guess it only probably worked before because at the beginning of the test it was paused.